Добавлен просмотр своего коливинга

NRenat opened the pull request:
- UserResidenceUpdateAPIView переименован в UserResidenceAPIView. Добавлена возможность получать коливинг по telegram_id

- Переименовны кнопки COLIVING_PROFILE_DUPLICATE_BUTTONS в COLIVING_PROFILE_FOR_OWNER_BUTTONS для лучшей читаемости.

- Добавлены COLIVING_PROFILE_BUTTONS для пользователй коливинга

- Добавлен States.COLIVING_CURRENT_USER для участников коливинга.

NRenat committed:
Добавлен просмотр своего коливинга

- Теперь можно просматривать свой коливинг.
- Клавиатура соответствует правам жильцов.

TomatoInOil commented on src/backend/profiles/api_views.py:
```diff
@@ -102,13 +103,25 @@ def get_queryset(self):
     serializer_class = UserResidenceSerializer
     lookup_field = "telegram_id"
+
+    def get(self, request, *args, **kwargs):
+        """Получить коливинг текущего проживающего."""
```
Определяя метод `.get()` ты заменяешь логику `generics.RetrieveAPIView`, то есть наследование от `generics.RetrieveAPIView` перестаёт иметь смысл. 
Также здесь происходит смешивание сущностей, обновляем мы сущность `User`, а получаем `Coliving`.
Можно использовать `RetrieveAPIView`, но отдельно от этого класса без переопределения метода `get()`.

```Python
# src/backend/profiles/api_views.py

class UserResidenceGetAPIView(generics.RetrieveAPIView):
    queryset = Coliving.objects.all()
    lookup_field = "roommates__telegram_id"
    lookup_url_kwarg = "telegram_id"
    serializer_class = ColivingSerializer
```

И зарегистрировать `url` так, чтобы было понятно, что мы получаем `Coliving`.
`...api/v1/users/<int:telegram_id>/residence/`

Здесь можно прочитать про `lookup_field` и `lookup_url_kwarg`:  https://www.django-rest-framework.org/api-guide/generic-views/#genericapiview

TomatoInOil replied:
Нужно добавить минимальные тесты для этой ручки.

TomatoInOil commented on src/bot/conversations/coliving/handlers.py:
```diff
@@ -286,6 +286,16 @@
                 callback=coliving_transfer.handle_cancel_coliving_transfer,
             ),
         ],
+        States.COLIVING_CURRENT_USER: [
+            CallbackQueryHandler(
```
Хорошо, что вынес в отдельный State

NRenat committed:
Ревью

- Перенесена логика получения коливинга по пользователю в отдельную CBV.
- Добавлен новый url, для получения коливинга пользователя по telegram_id.
- Добавлен тест для нового url.
- Методу _get_request в service.py добавлена обработка ошибок. 404 теперь возвращается ответом для последующей обработки.

TomatoInOil committed:
Resolve conflicts.

# Conflicting files
#    src/bot/conversations/coliving/states.py
